pkg/oc/cli/cmd/set/deployment_strategy.go

```diff
@@ -0,0 +1,483 @@
+package set
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/kubernetes/pkg/apis/apps"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+
+	deployapi "github.com/openshift/origin/pkg/apps/apis/apps"
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+var (
+	deploymentStrategyLong = templates.LongDesc(`
+		Set a deployment strategy for a deployment config, deployment,
+		statefulset and daemonset.
+
+`)
+
+	deploymentStrategyExample = templates.Examples(`
+		# Change the deployment strategy for dc/myapp to recreate
+	  %[1]s deployment-strategy dc/myapp --recreate
+
+		# Set the max-surge option for deployment config
+	  %[1]s deployment-strategy dc/myapp --rolling --max-surge=2
+
+		# Change the deployment strategy to OnDelete for a statefulset and sets the partiion
+	  %[1]s deployment-strategy ss/myapp --rolling --partition=10
+
+`)
+)
+
+// DeploymentStrategyOptions contain options for set deployment-strategy.
+type DeploymentStrategyOptions struct {
+	Out         io.Writer
+	Err         io.Writer
+	Builder     *resource.Builder
+	Infos       []*resource.Info
+	Encoder     runtime.Encoder
+	Filenames   []string
+	Output      string
+	Mapper      meta.RESTMapper
+	Selector    string
+	ShortOutput bool
+	Local       bool
+
+	PrintObject func([]*resource.Info) error
+	Cmd         *cobra.Command
+
+	// DeploymentConfig and Deployment
+	Recreate bool
+	Rolling  bool
+	Custom   bool
+
+	// StatefulSet and DaemonSet
+	OnDelete bool
+
+	// DeploymentConfig strategies options
+	TimeoutSeconds      int64
+	UpdatePeriodSeconds int64
+	IntervalSeconds     int64
+	MaxUnavailable      string
+	MaxSurge            string
+	Image               string
+	Command             []string
+	Environment         []string
+
+	// StatefulSet
+	Partition int32
+}
+
+// NewCmdDeploymentStrategy implements the set deployment-strategy command
+func NewCmdDeploymentStrategy(fullName string, f *clientcmd.Factory, out, errOut io.Writer) *cobra.Command {
+	options := &DeploymentStrategyOptions{
+		Out: out,
+		Err: errOut,
+	}
+	cmd := &cobra.Command{
+		Use:     "deployment-strategy RESOURCE --recreate|--rolling|--on-delete|--custom",
+		Short:   "Update a deployment strategy",
+		Long:    deploymentStrategyLong,
+		Example: fmt.Sprintf(deploymentStrategyExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
```

Comment:
Why don't we use RunE and check errors and exit at one place?

Reply:
this copy the pattern from set deploy hook, I can change that here but I guess I will prefer switch in other commands as well? // @fabianofranz @juanvallejo

Reply:
@mfojtik This could be tracked in a separate issue, preferably upstream, since we would want to begin using RunE there first